src/nodes/ImageView.vue

```diff
@@ -295,6 +301,15 @@ export default {
 			tr.delete(pos, pos + this.node.nodeSize)
 			this.view.dispatch(tr)
 		},
+		async setImageSrc(imageUrl) {
+			const file = await axios.get(imageUrl)
```

Comment:
To avoid getting the whole image data before knowing if the mime type can actually be handled, we could use a HEAD request which also returns the `Content-Type` header.

Reply:
Is this what you have in mind? 5ea058b94bc6a3404a292244f81abce9690b0a51

Now I feel like the `imageUrl2` trick could be removed and image loading could be clarified. How about doing this: 5ffccc411e8838cbc52ac8c2d025b3c8375bed68 ? (pushed in another branch for the moment). @max-nextcloud I think it's close to what you described.

The logic is to always load the image with an `Image()` object. If it's successful, we set the `<img>` src attribute. As the image has already been loaded, the browser does not download it again.

There are 3 major cases:
* (1) data URL: we directly load it in the `Image()`
* not a data URL: we get the mime type with a HEAD request
    * (2) SVG: we download it, generate a base64 string and load it in the `Image()`
    * (3) other mime types: we directly use the URL in the `Image()`

Reply:
I've been talking about this with Max earlier today, and we could probably even avoid the HEAD request and use the GET, if browsers are able to properly cache that and then use the cached image when setting the src of the img-tag